src/libutil/rust-ffi.hh

```diff
@@ -0,0 +1,84 @@
+#include "serialise.hh"
+
+namespace rust {
+
+// Depending on the internal representation of Rust slices is slightly
+// evil...
```

Comment:
It is. There's zero guarantee that the internal representation of a slice will look like this.